infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/PodMerger.java

```diff
@@ -116,6 +116,11 @@ public Deployment merge(List<PodData> podsData) throws ValidationException {
         }
       }
 
+      baseSpec.setTerminationGracePeriodSeconds(
```

Comment:
IMO, we should always fall back on `terminationGracePeriodSeconds` aka baseSpec.